### DISCUSSION TITLE
Is there a rule that always forces single line dictionaries?

### DISCUSSION THREAD

**Discussion by Kludex** [astral-sh/ruff#13447] [Q&A]
Like, I don't want this:

```py
{
    "potato": "value",
    "key": "value",
}
```
I always want this:
```py
{"potato": "value", "key": "value"}
```

Is there a rule for this?

**Comment by MichaReiser** [comment]
No, I don't think such rule exists. You can get this result when using `ruff format` (be warned, it applies many other formatting rules as well) when setting `skip-magic-trailing-comma=true`. But only if you're okay if it splits the dictionary over multiple lines if it exceeds the configured line length

**Reply by Kludex** [reply]
That will do. Thanks.

**Comment by Kludex** [comment]
Okay, sorry, I'm back. Let me refine my request...

I would like to try to enforce this if it's possible (`max-length` allows):
```py
{"potato": "value", "key": "value"}
```
But then if it's not possible, I'd like to enforce this (multiline with trailing commas):
```py
{
    "potato": "value",
    "key": "value",
}
```

I hate this:
```py
{
    "potato": "value", "key": "value"
}
```

**Reply by MichaReiser** [reply]
You can use COM812 together with `ruff format`. `COM812` adds a trailing comma after the second entry, the formatter then splits the dictionary across multiple lines ([Playground](https://play.ruff.rs/2a6f08b3-aa43-419f-87f9-1cdee3d54c00))

However, this only works when **not** using `skip-magic-trailing-comma`. Otherwise, arguments might get formatted as 

```python
def test(
    aaaaaaa, bbbbbbbbbb, cccccccccccccc, dddddddddddddddddddddddddddddddddddddddddd
):
    pass
```

which is probably not to your taste ;) 

However, if you use `skip-magic-trailing-comma`, then lines won't get collapsed if they now fit into the line length. There's currently no option that allows you to have both.
